### DISCUSSION TITLE
Surfaces are not rendered correctly sometimes

### DISCUSSION THREAD

**Discussion by Chrisomator** [pyvista/pyvista#3544] [Q&A] [answered]
Hello guys,

thank you for this awesome project! I'm trying to use it to render several CAD models from different angles in python.

When I use the "normal" render window with Plotter.show() or with the QT Version and turn the model around with my mouse, I've got no issues with my models.

But when I try to render the model with off_screen=True and rotate the mesh of the model via rotate_x, rotate_y or rotate_z, sometimes some surfaces are not rendered correctly, mostly edges are affected.

I have already tried to call p.update() after I rotated the mesh, but it doesn't help.

Do you have any ideas, what could cause this issues?

Thanks in advance!
![240](https://user-images.githubusercontent.com/7378413/199604919-7d9b45e6-8b19-4c71-87bf-a5149d074ad2.png)
![255](https://user-images.githubusercontent.com/7378413/199604921-123e91f1-9c7c-4b86-b5aa-5920410a6741.png)
<img width="511" alt="renderwindow" src="https://user-images.githubusercontent.com/7378413/199605225-0f9337d9-5fa2-496a-9978-716a2a542b02.png">

**Comment by banesullivan** [comment, marked as answer]
Can you share the mesh and code to reproduce the issue?

Otherwsie, If using rotate_x, rotate_y or rotate_z on the mesh directly and off_screen rendering, then I suspect the renderer is failing to re-render/update which would have the effect of using the wrong camera clipping planes. I recommend calling `plotter.update()` and/or `plotter.camera.reset_clipping_range()`

**Reply by Chrisomator** [reply]
Hey! Thank you for your response. I have changed the clipping range to a huge value (100, 1000) and it fixed the problem. Thanks a lot.
